src/hooks/useCart/useCart.test.tsx

```diff
@@ -182,152 +213,20 @@ describe("useCart", () => {
-      expect(result.current.cartState).toBe("FETCHING_QUOTA");
-
-      await waitForNextUpdate();
-      expect(result.current.cartState).toBe("NO_QUOTA");
-      expect(result.current.cart).toStrictEqual([
-        {
-          category: "toilet-paper",
-          descriptionAlert: undefined,
-          identifierInputs: [],
-          lastTransactionTime: transactionTime,
-          maxQuantity: 0,
-          quantity: 0
-        },
-        {
-          category: "chocolate",
-          descriptionAlert: undefined,
-          identifierInputs: [],
-          lastTransactionTime: transactionTime,
-          maxQuantity: 0,
-          quantity: 0
-        }
-      ]);
-    });
-    it("should have cart state be NO_QUOTA when quota received is invalid", async () => {
-      expect.assertions(3);
-      mockGetQuota.mockReturnValueOnce(mockQuotaResSingleIdInvalidQuota);
-
-      const ids = ["ID1"];
-      const { result, waitForNextUpdate } = renderHook(
-        () => useCart(ids, key, endpoint),
-        { wrapper }
-      );
-      expect(result.current.cartState).toBe("FETCHING_QUOTA");
-
-      await waitForNextUpdate();
-      expect(result.current.cartState).toBe("NO_QUOTA");
-      expect(result.current.cart).toStrictEqual([
-        {
-          category: "toilet-paper",
-          descriptionAlert: undefined,
-          identifierInputs: [],
-          lastTransactionTime: transactionTime,
-          maxQuantity: 0,
-          quantity: 0
-        },
-        {
-          category: "chocolate",
-          descriptionAlert: undefined,
-          identifierInputs: [],
-          lastTransactionTime: transactionTime,
-          maxQuantity: 15,
-          quantity: 0
-        }
-      ]);
-    });
-    it("should set cart state to be NOT_ELIGIBLE when NotEligibleError is thrown, and would not continue with fetching quota", async () => {
-      expect.assertions(1);
-
-      const ids = ["ID_NOT_ELIGIBLE"];
-
-      mockGetQuota.mockImplementation(() => {
-        mockIdNotEligible(ids[0]);
-      });
-
-      const { result } = renderHook(() => useCart(ids, key, endpoint), {
-        wrapper
-      });
-
-      expect(result.current.cartState).toBe("NOT_ELIGIBLE");
-    });
-  });
-
   describe("update cart quantities", () => {
-    it("should update the cart when more ids are added", async () => {
+    it("should update the cart when quota response changes", () => {
       expect.assertions(1);
-      mockGetQuota.mockReturnValueOnce(mockQuotaResSingleId);
 
       let ids = ["ID1"];
-      const { rerender, result, waitForNextUpdate } = renderHook(
-        () => useCart(ids, key, endpoint),
+      const { rerender, result } = renderHook(
+        () =>
+          useCart(ids, key, endpoint, mockQuotaResMultipleIds.remainingQuota),
         { wrapper }
       );
 
-      await waitForNextUpdate();
-
-      mockGetQuota.mockReturnValueOnce(mockQuotaResMultipleIds);
       ids = ["ID1", "ID2"];
-      rerender([ids, key, endpoint]);
```

Comment:
Right, this is actually the point on upgrading react hooks testing lib, we can't do this anymore. Looks like it really was the useCart tests lolol @yongjiajie